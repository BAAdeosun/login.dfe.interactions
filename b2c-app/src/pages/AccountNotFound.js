import React from 'react';
import components from '../components';
import { ACTIONS } from '../constants/actions';

class AccountNotFound extends React.Component {

    componentDidMount() {
        document.getElementById('api').style.display = 'none';
        document.title = 'We have not been able to find your account | National Careers Service';
    }

    render() {

        const resetPasswordParagraph = [
            <components.Link action={ACTIONS.RESET_PASSWORD} text="Try again" key="retry" />,
            " with other details you may have used when you created your account."
        ];

        const createNewAccountParagraph = [
            "If you're unable to recover your registered email address, you'll have to ",
            <components.Link action={ACTIONS.SIGNUP} text="create a new account" key="signup" />,
            "."
        ];

        const content = [
            <components.Paragraph text={resetPasswordParagraph} key='paragraph1' />,
            <components.Paragraph text={`If you think you could have more than one account, call 0800 100 900. We'll confirm your details and delete any accounts you do not need.`} key='paragraph2' />,
            <components.Paragraph text={createNewAccountParagraph} key='paragraph3' />
        ];

        return (

            <div id="accountNotFound">
                <components.PageContainer
                    pageTitle="We have not been able to find your account"
                    aboveFormContent={content}
                />
            </div>
        )
    }
}

export default AccountNotFound;
