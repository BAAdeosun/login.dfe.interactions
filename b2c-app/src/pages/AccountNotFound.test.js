import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AccountNotFound from './AccountNotFound';

it('renders without crashing', () => {
    shallow(<AccountNotFound />);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<AccountNotFound />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});