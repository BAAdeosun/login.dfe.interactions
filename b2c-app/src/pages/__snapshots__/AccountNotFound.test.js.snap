// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="accountNotFound"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              We have not been able to find your account
            </h1>
            <div>
              <p
                className="govuk-body"
              >
                <a
                  className="govuk-link"
                  href={null}
                  onClick={null}
                >
                  Try again
                </a>
                Â with other details you may have used when you created your account.
              </p>
              <p
                className="govuk-body"
              >
                If you think you could have more than one account, call 0800 100 900. We'll confirm your details and delete any accounts you do not need.
              </p>
              <p
                className="govuk-body"
              >
                If you're unable to recover your registered email address, you'll have to create a new account.
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
